src/diagnostics.ts

```diff
@@ -27,14 +27,16 @@ export function convertTsDiagnostic(diagnostic: ts.Diagnostic): Diagnostic {
  *
  * @param category The Typescript DiagnosticCategory
  */
-function convertDiagnosticCategory(category: ts.DiagnosticCategory): DiagnosticSeverity {
+function convertDiagnosticCategory(category: ts.DiagnosticCategory): DiagnosticSeverity | undefined {
     switch (category) {
         case ts.DiagnosticCategory.Error:
             return DiagnosticSeverity.Error
         case ts.DiagnosticCategory.Warning:
             return DiagnosticSeverity.Warning
         case ts.DiagnosticCategory.Message:
             return DiagnosticSeverity.Information
+        default:
```

Comment:
This shouldn't have a default, the fact that TS will error if not all DiagnosticSeverities are handled is a good thing. This should instead handle the new DiagnosticSeverity. This should map to `Hint`